mirdata/irmas.py

```diff
@@ -0,0 +1,349 @@
+    """Load a IRMAS dataset audio file.
+
+    Args:
+        audio_path (str): path to audio file
+
+    Returns:
+        y (np.ndarray): the mono audio signal
+        sr (float): The sample rate of the audio file
+
+    """
+    if not os.path.exists(audio_path):
+        raise IOError("audio_path {} does not exist".format(audio_path))
+    return librosa.load(audio_path, sr=44100, mono=False)
+
+
+def download(
+    data_home=None, partial_download=None, force_overwrite=False, cleanup=True
+):
+    """Download the IRMAS Dataset.
+
+    Args:
+        data_home (str):
+            Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+        force_overwrite (bool):
+            Whether to overwrite the existing downloaded data
+        partial_download (list):
+            List indicating what to partially download. The list can include any of:
+                * 'TODO_KEYS_OF_REMOTES' TODO ADD DESCRIPTION
+            If `None`, all data is downloaded.
+        cleanup (bool):
+            Whether to delete the zip/tar file after extracting.
+
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    download_utils.downloader(
+        data_home,
+        remotes=REMOTES,
+        partial_download=partial_download,
+        info_message='',
+        force_overwrite=force_overwrite,
+        cleanup=cleanup,
+    )
+
+
+def validate(data_home=None, silence=False):
+    """Validate if the stored dataset is a valid version
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        missing_files (list): List of file paths that are in the dataset index
+            but missing locally
+        invalid_checksums (list): List of file paths that file exists in the dataset
+            index but has a different checksum compare to the reference checksum
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    missing_files, invalid_checksums = utils.validator(
+        DATA.index, data_home, silence=silence
+    )
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    """Return track ids
+
+    Returns:
+        (list): A list of track ids
+    """
+    return list(DATA.index.keys())
+
+
+def load(data_home=None):
+    """Load IRMAS dataset
+
+    Args:
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets`
+    Returns:
+        (dict): {`track_id`: track data}
+    """
+    if data_home is None:
+        data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+    data = {}
+    for key in DATA.index.keys():
+        data[key] = Track(key, data_home=data_home)
+    return data
+
+
+def load_pred_inst(audio_path, annotation_path, train):
```

Comment:
maybe I missed some discussion before, but isn't this supposed to be the folder where the audio file corresponding to current Track is stored? 
this can be obtained easily with 
`os.path.basename(os.path.dirname(audio_path))`

Reply:
Quick comment here - the load_* functions should be as 'generic' as possible. I don't know this dataset well, but I don't think it's good to be passing `audio_file` just to parse the path. Is it the case that test tracks have annotation files, while training tracks don't, and the label is inferred from the filename? What is the difference between the test track file names and the test track annotations?

It looks like during training, you use the name of the file to parse the predominant instrument - to me this should just be in the track metadata, not part of this load function. During testing, it loads an annotation file and this part seems perfectly fine. If there are no annotation files for some tracks, we have other loaders that return e.g. `None` when the file does not exist (see `salami.py` for some examples)

My proposal is this function should just have:
```
def load_pred_inst(annotation_path):
    with open(annotation_path):
        ...
    # plus some possible check for if annotation_path is None
```

Then, in `Track.__init__` you could create an attribute `self._predominant_instrument=audio_path.split('[')[1].split(']')[0]` 

In `self.instrument`, you can return `self._predomiant_instrument` if it's set, and load from the annotation otherwise.

Reply:
Ok, done!! The only issue here is that now Track has `instrument` with the list of strings as the "official" annotation, and `predominant_instrument` which is the attribute I added to `Track.__init__`. Is this Ok @rabitt? Thanks for the feedback:)